engine/Simulator/SimulateMain.py

```diff
@@ -17,66 +17,53 @@ def simulate_main():
 	logger.debug("Calling simulate_main()...")
 
 	for idx, statement in enumerate(test_cases, start=1):
-		logger.debug("Processing statement=%s", statement)
-		# print(f"\n\nTest Case {idx}: Processing statement '{statement}'...")
+		logger.info("Statement:\t%s", statement)
 
 		# First complete all checks
 		try:
-			logger.info("Before proceeding, checks for unequal brackets and "
-						"illegal characters are done.")
+			logger.info("Before proceeding, doing checks for unequal brackets and "
+						"illegal characters.")
 			if (check_left_and_right_brackets(statement) and
 					check_for_illegal_characters(statement)):
 				pass
 		except UnequalBracketsExcept as ce:
-			logger.error("Unequal Brackets Exception: ", ce)
+			logger.exception("Unequal Brackets Exception: ", ce)
 			continue
 		except IllegalCharactersException as ice:
-			logger.error("Illegal Characters Exception: ", ice)
+			logger.exception("Illegal Characters Exception: ", ice)
 			continue
 		else:
-			# print("No exception occurred!")
 			pass
 
 		# todo Look into why statement is sent to user_input() and then
 		#  returned again
 
-		logger.warning("All checks are valid, proceeding with extracting all "
-					"needed information.")
+		logger.info("All checks are valid, proceeding with extracting "
+					"information from statement.")
 		number_of_variables, variables_as_array, provided_statement, elements_in_tree = (
 			user_input(statement))
 		logger.debug("In simulate_main(): \nnumber_of_variables=%d, \nvariables_as_array=%s, "
 					 "\nprovided_statement=%s, \nelements_in_tree=%s",
 					 number_of_variables, variables_as_array, provided_statement, elements_in_tree)
 
-		print_details(number_of_variables, variables_as_array, provided_statement, elements_in_tree)
+		print_details("SUMMARY",
+					  ("number_of_variables", number_of_variables),
+					  ("variables_as_array", variables_as_array),
+					  ("provided_statement", provided_statement),
+					  ("elements_in_tree", elements_in_tree))
 
-
-		# logger.warning("Starting with evaluation of array...")
 		final_value_array, returned_matrix, all_variables = (
-			evaluate_statement_as_array(number_of_variables, variables_as_array, elements_in_tree))
-		logger.warning("Finished evaluating statement.")
-		logger.debug("final_value_array=%s, returned_matrix=%s, all_variables=%s", final_value_array, returned_matrix, all_variables)
+			evaluate_array_as_tree(number_of_variables, variables_as_array, elements_in_tree))
+		logger.debug("In simulate_main(): \nfinal_value_array=%s, \nreturned_matrix=%s, \nall_variables=%s",
+					 final_value_array, returned_matrix, all_variables)
+
+		print_details("Evaluate", ("Statement", statement),
```

Comment:
I believe the rule is to use capital letters in the first parameter? Please confirm.

Reply:
Yes, that is correct. Changed.